### PR TITLE
Fix/dataset size error

### LINKED ISSUE #22: Kullback-Leibler Loss without dataset size should raise Error
The Kullback-Leibler Loss implementation tries to infer the dataset size from the second element of the inputs shape (which is assumed to be the batch dimension). Since this behavior is not consistent for arbitrary ordering of dimensions and the dataset size is an essential parameter to balance data fitting and prior matching term correctly, this behavior should be replaced with an appropriate error message.

### PR DESCRIPTION
**Pull request by suryaramku**
Closes #22

The KL loss functions (`KullbackLeiblerLoss` and `AnalyticalKullbackLeiblerLoss`) previously issued a warning and inferred `dataset_size` from the batch dimension when not provided. This behavior is unreliable. 

This PR replaces that fallback with a `ValueError` requiring the user to provide `dataset_size` either in the constructor or in the `forward` call.

Changes:
- `torch_blue/vi/kl_loss.py`: warning → ValueError
- `torch_blue/vi/analytical_kl_loss.py`: same change
- Updated tests to expect ValueError instead of UserWarning

### CODE REVIEW

File: tests/test_vi/test_analytical_kl_loss.py [hunk 1]
```diff
@@ -765,10 +765,17 @@ def test_forward(
...
-            message = f"No dataset_size is provided. Batch size \\({batch_size}\\) is used instead."
-            with pytest.warns(UserWarning, match=message):
-                analytical_loss = criterion(output, target, fwrd_dataset_size)
+            with pytest.raises(
+                ValueError,
+                match="dataset_size must be provided",
+            ):
+                criterion(output, target, fwrd_dataset_size)
             ref_loss = ref_criterion(output, target, batch_size)
+            # Skip the rest of this iteration since forward raised
```
Comment (ArvidWeyrauch) [inline tests/test_vi/test_analytical_kl_loss.py:774]: This test could be streamlined slightly by just putting an early ''return'' here and removing the ref loss and everything until and including the continue

**Review by ArvidWeyrauch — CHANGES_REQUESTED** [review]
Thank you for your interest in torch-blue and taking this issue off our to-list. I left one comment for a minor streamline in test_analytical_kl_loss.py (which I am happy to do myself if you are busy with other stuff, but it felt rude to commit to your fork without giving you the chance to do it yourself)
Thank you for the contribution, will be merged once the streamline is in
